Source/TheAviator/flights/FlightReporter.ts

```diff
@@ -4,15 +4,77 @@
 import { IFlightReporter } from './IFlightReporter';
 import { ScenarioResult } from './reporting';
 import { Flight } from './Flight';
-import { Scenario, ScenarioEnvironment } from '../gherkin';
+import { Scenario, ScenarioEnvironment, RunState, BecauseOf } from '../gherkin';
 
 import chalk from 'chalk';
+import { merge, Observable } from 'rxjs';
+import { Transform, TransformCallback } from 'stream';
+import { Console } from 'console';
+
+class NewlineCountingStream extends Transform {
+    newlines: number = 0;
+
+    _transform(chunk: any, encoding: BufferEncoding, callback: TransformCallback) {
+        const chunkNewlines = (chunk.toString().match(/\n/g) || []).length as number;
+        this.newlines += chunkNewlines;
+        this.push(chunk, encoding);
+        callback();
+    }
+}
+
+class NewlineCounter {
+    readonly _stream: NewlineCountingStream;
+
+    constructor() {
+        this._stream = new NewlineCountingStream();
+        this._stream.pipe(process.stdout);
+    }
+
+    get console(): Console {
+        return new Console(this._stream);
+    }
+
+    get newlines(): number {
+        return this._stream.newlines;
+    }
+
+    destroy() {
+        this._stream.unpipe()
+        this._stream.destroy();
+    }
+}
+
+type OverwritingConsoleRendererCallback = (console: Console) => void;
+
+class OverwritingConsoleRenderer {
```

Comment:
Same as above - own file.
You can include the delegate type in the same as well.